File: src/Data_Structures/Stack/stack.ts
import { LinkedList } from "../Linked_List/linked-list";

export class Stack<T> {
  private list: LinkedList<T>;

  constructor(...items: T[]) {
    this.list = new LinkedList();
    items.map((item) => this.list.push(item));
  }

  push(...items: T[]) {
    items.map((item) => this.list.push(item));
    return this.list.length;
  }

  pop() {
    return this.list.pop();
  }

  peek() {
    return this.list.tail?.value;
  }

  get length() {
    return this.list.length;
  }

  isEmpty() {
    return this.list.isEmpty();
  }

  toArray() {
    return this.list.toArray();
  }
}

File: src/Data_Structures/Stack/stack.py
from src.Data_Structures.Linked_List.linked_list import LinkedList


class Stack:
    def __init__(self, *items):
        self.list = LinkedList()
        for item in items:
            self.list.push(item)

    def push(self, *items):
        for item in items:
            self.list.push(item)

        return self.list.length

    def pop(self):
        return getattr(self.list.pop(), 'value', None)

    def peek(self):
        return getattr(self.list.tail, 'value', None)

    def get_length(self):
        return self.list.length

    def is_empty(self):
        return self.list.is_empty()

    def to_list(self):
        return self.list.to_list()
